metadata-ingestion/tests/integration/snowflake/test_snowflake_tag.py

```diff
@@ -0,0 +1,103 @@
+            account_id="ABC12345.ap-south-1.aws",
+            username="TST_USR",
+            password="TST_PWD",
+            match_fully_qualified_names=True,
+            schema_pattern=AllowDenyPattern(allow=["test_db.test_schema"]),
+            tag_pattern=AllowDenyPattern(
+                allow=["TEST_DB.TEST_SCHEMA.my_tag_1:my_value_1"]
+            ),
+            include_technical_schema=True,
+            include_table_lineage=False,
+            include_view_lineage=False,
+            include_column_lineage=False,
+            include_usage_stats=False,
+            include_operational_stats=False,
+            extract_tags=TagOption.without_lineage,
+        )
+
+        pipeline = Pipeline(
+            config=PipelineConfig(
+                source=SourceConfig(type="snowflake", config=tag_config),
+                sink=DynamicTypedConfig(type="blackhole", config={}),
+            )
+        )
+        pipeline.run()
+        pipeline.pretty_print_summary()
+        pipeline.raise_from_status()
+
+        source_report = pipeline.source.get_report()
+        assert isinstance(source_report, SnowflakeV2Report)
+        assert cast(SnowflakeV2Report, source_report).tags_scanned == 5
+        assert cast(SnowflakeV2Report, source_report)._processed_tags == {
+            "TEST_DB.TEST_SCHEMA.my_tag_1:my_value_1"
+        }
+
+
+def test_snowflake_tag_pattern_deny():
+    with mock.patch("snowflake.connector.connect") as mock_connect:
+        sf_connection = mock.MagicMock()
+        sf_cursor = mock.MagicMock()
+        mock_connect.return_value = sf_connection
+        sf_connection.cursor.return_value = sf_cursor
+        sf_cursor.execute.side_effect = default_query_results
+
+        tag_config = SnowflakeV2Config(
+            account_id="ABC12345.ap-south-1.aws",
+            username="TST_USR",
+            password="TST_PWD",
+            match_fully_qualified_names=True,
+            schema_pattern=AllowDenyPattern(allow=["test_db.test_schema"]),
+            tag_pattern=AllowDenyPattern(
+                deny=["TEST_DB.TEST_SCHEMA.my_tag_2:my_value_2"]
+            ),
+            include_technical_schema=True,
+            include_table_lineage=False,
+            include_view_lineage=False,
+            include_column_lineage=False,
+            include_usage_stats=False,
+            include_operational_stats=False,
+            extract_tags=TagOption.without_lineage,
+        )
+
+        pipeline = Pipeline(
+            config=PipelineConfig(
+                source=SourceConfig(type="snowflake", config=tag_config),
+                sink=DynamicTypedConfig(type="blackhole", config={}),
+            )
+        )
+        pipeline.run()
+        pipeline.pretty_print_summary()
+        pipeline.raise_from_status()
+
+        source_report = pipeline.source.get_report()
+        assert isinstance(source_report, SnowflakeV2Report)
+        assert cast(SnowflakeV2Report, source_report).tags_scanned == 5
```

Comment:
after the `assert isinstance`, I believe the `cast` call is redundant. mypy understands that the isinstance call does type narrowing

Reply:
thanks! i didn't know what it did and i was copying from other tests hehe